### PR TITLE
Speedup generateRules()

### PR DESCRIPTION
**Pull request by laurentheirendt**
Speedup by a factor of 3-4, depending on the model. This is a serial speedup - no parallelization.

For instance, for the model `modelR204` loaded from `Recon2.v04.mat`:
- **Existing version:**
```Matlab
>> tic; newModel = generateRules(modelR204); toc;
Elapsed time is 14.444188 seconds
```

- **New version:**
```Matlab

>> tic; newModel = generateRules(modelR204); toc;
Elapsed time is 4.584043 seconds.
```

**I hereby confirm that I have:**

- [x] Tested my code on my own machine
- [x] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [x] Selected `develop` as a target branch (top left drop-down menu)

### CODE REVIEW

File: src/base/parseGPR.m [hunk 1]
```diff
@@ -8,51 +8,61 @@
...
 % whitespace character or a opening or closing bracket, respectively. Gene
 % Names can also be at the beginning or the end of the string.
-% 
-% 
+%
+%
 % USAGE:
 %
-%    [newGeneList,totalGeneList,ruleString] = generateRules(grRuleString,currentGenes)
+%    [newGeneList,totalGeneList,ruleString] = generateRules(grRuleString,totalGeneList)
```
Comment (tpfau) [inline src/base/parseGPR.m]: The header still says `currentGenes`

Comment (laurentheirendt) [reply src/base/parseGPR.m]: true 😄

File: src/base/parseGPR.m [hunk 2]
```diff
@@ -8,51 +8,61 @@
...
 %
 % INPUT:
 %    grRuleString:     The rule string in textual format.
```
Comment (tpfau) [inline src/base/parseGPR.m]: Also this is no longer true. If its a cell array now, it expects a different input, maybe this should be marked with an additional flag?

Comment (laurentheirendt) [reply src/base/parseGPR.m]: What exactly is no longer true?

Comment (tpfau) [reply src/base/parseGPR.m]: Essentially I'm wondering about line 40:
`if ~iscell..`
Why is this check necessary?

Comment (laurentheirendt) [reply src/base/parseGPR.m]: This check is necessary because of lines 20 and 31 of `generateRules`.

**Comment by syarra** [comment]
Nice speedup! Thanks @laurentheirendt 🍰

**Pull request merged**